cpp/src/holtwinters/HoltWinters.cpp

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_runtime.h>
+
+#include <iostream>
+#include "HoltWinters.hpp"
+#include "holtwinters_utils.hpp"
+#include "hw_cu_utils.hpp"
+#include "utils.h"
+
+ML::HWStatus ML::HWInit() {
+  ML::cublas::get_handle();
+  ML::cusolver::get_handle();
+  return ML::HWStatus::HW_SUCCESS;  // TODO(ahmad): check cublas
+}
+
+ML::HWStatus ML::HWDestroy() {
+  ML::cublas::destroy_handle();
+  ML::cusolver::destroy_handle();
+  return ML::HWStatus::HW_SUCCESS;  // TODO(ahmad): check cublas
+}
+
+template <typename Dtype>
+ML::HWStatus ML::HWTranspose(const Dtype *data_in, int m, int n,
+                             Dtype *data_out) {
+  if (!data_in || !data_out || n < 1 || m < 1)
+    return ML::HWStatus::HW_INVALID_VALUE;
+
+  return transpose_gpu(data_in, m, n, data_out);
```

Comment:
I understand this will be in a separate PR, but pointing out here that we have a transpose prim.